compiler/plugins/input/StableHLO/Conversion/test/auto_input_conversion.mlir

```diff
@@ -8,3 +8,13 @@ func.func @simple_add_stablehlo(%arg0: tensor<2x2xi32>, %arg1: tensor<2x2xi32>)
   %0 = stablehlo.add %arg0, %arg1 : (tensor<2x2xi32>, tensor<2x2xi32>) -> tensor<2x2xi32>
   return %0 : tensor<2x2xi32>
 }
+
+// -----
+
+// CHECK-LABEL: util.func public @vhlo_func
+vhlo.func_v1 @vhlo_func(%arg0: !vhlo.tensor_v1<!vhlo.i32_v1>) -> (!vhlo.tensor_v1<!vhlo.i32_v1>) {
+  // CHECK: arith.constant
+  %0 = "vhlo.constant_v1"() <{value = #vhlo.tensor_v1<dense<1> : tensor<i32>>}> : () -> !vhlo.tensor_v1<!vhlo.i32_v1>
+  // CHECK: return
+  "vhlo.return_v1"(%0) : (!vhlo.tensor_v1<!vhlo.i32_v1>) -> ()
```

Comment:
What happens with vhlo and stablehlo ops are mixed in the source IR? Is this allowed? Could we have a test that makes sure this is either supported or rejected?

Reply:
Could you add a test that exercises this and documents the expected behavior?

Reply:
I am sorry but I don't know how to do that (or maybe I am not understanding what you are asking 😄 ). Having both in the same is considered "undefined behavior" and if you mean having a test with no VHLO then just any stablehlo test would suffice to test that.

I will add a comment saying like this right above the call:

```
  // if the input is VHLO file, then it is automatically converted to Stablehlo
  // otherwise, this Pass is considered a NOP.
  // Note: Having StableHLO and VHLO in the same function is considered
  // undefined behavior.
  ::mlir::stablehlo::createStablehloDeserializePipeline(passManager);
```

Reply:
We often see bug reports opened by external contributors who end up with mlir inputs that mix dialects. That's why it's important to detect any unsupported inputs and produce clear errors about it.

Even if this is left undefined by vhlo, it's not like a runtime integer overflow in C that's difficult to detect; here, the behavior is entirely static and the check itself shouldn't be complicated. I would like us to have a test that mixes vhlo and stablehlo and see a couple options:

1. If this errors out within the vhlo deserialization passes, catch that with an expected error.
2. If this works with vhlo deserialization passes, we can have a simple iree pass to check for inputs that mix stablehlo and vhlo. IIRC, we should already have something similar that checks that there are no remaining stablehlo ops after at the end of input conversion. This should produce an error that explains that mixing the two is not supported in IREE, and the test should check for this error.

I'd be also interested to learn how this interacts with other dialects like chlo, and have this captured as a test.

Reply:
Added a test to mix vhlo and stablehlo that the compiler rejects ([db0acc3](https://github.com/openxla/iree/pull/16999/commits/db0acc36707187cca5d57d31f3378fced33352a6))